src/main/java/com/opentable/db/postgres/embedded/EmbeddedPostgreSQL.java

```diff
@@ -316,29 +313,23 @@ private void cleanOldDataDirectories(File parentDirectory)
             if (!lockFile.exists() || isTooNew) {
                 continue;
             }
-            try {
-                final FileOutputStream fos = new FileOutputStream(lockFile);
-                try {
-                    try (FileLock lock = fos.getChannel().tryLock()) {
-                        if (lock != null) {
-                            LOG.info("Found stale data directory {}", dir);
-                            if (new File(dir, "postmaster.pid").exists()) {
-                                try {
-                                    pgCtl(dir, "stop");
-                                    LOG.info("Shut down orphaned postmaster!");
-                                } catch (Exception e) {
-                                    if (LOG.isDebugEnabled()) {
-                                        LOG.warn("Failed to stop postmaster " + dir, e);
-                                    } else {
-                                        LOG.warn("Failed to stop postmaster " + dir + ": " + e.getMessage());
-                                    }
-                                }
+            try (final FileOutputStream fos = new FileOutputStream(lockFile);
+                 final FileLock lock = fos.getChannel().tryLock()) {
+                if (lock != null) {
+                    LOG.info("Found stale data directory {}", dir);
+                    if (new File(dir, "postmaster.pid").exists()) {
+                        try {
+                            pgCtl(dir, "stop");
+                            LOG.info("Shut down orphaned postmaster!");
+                        } catch (Exception e) {
+                            if (LOG.isDebugEnabled()) {
+                                LOG.warn("Failed to stop postmaster " + dir, e);
+                            } else {
+                                LOG.warn("Failed to stop postmaster " + dir + ": " + e.getMessage());
                             }
-                            FileUtils.deleteDirectory(dir);
                         }
                     }
-                } finally {
-                    fos.close();
+                    FileUtils.deleteDirectory(dir);
```

Comment:
I wonder if it could make sense to wrap the directory deletion into an `AutoCloseable`.  Probably overkill.